notebook/terminal/handlers.py

```diff
@@ -18,6 +19,21 @@ def get(self, term_name):
                    ws_path="terminals/websocket/%s" % term_name))
 
 
+class NewTerminalHandler(IPythonHandler):
+    """Renders a new terminal interface using the named argument."""
+    @web.authenticated
+    def get(self, term_name):
+        new_path = self.request.path.replace("new/{}".format(term_name), term_name)
```

Comment:
If the `term_name` isn't set (i.e. they visit `/new`), can we have it create a new named one?

Reply:
Good point.  I added a commit supporting this.  This does preclude someone from creating their own named terminal of 'new' (via `GET /terminals/new/new`), but that's probably okay.